etcdserver/api/membership/cluster.go

```diff
@@ -252,6 +252,24 @@ func (c *RaftCluster) Recover(onSet func(*zap.Logger, *semver.Version)) {
 	}
 }
 
+// IsPromoteChange checks whether cc is a promote change
+func (c *RaftCluster) IsPromoteChange(cc raftpb.ConfChange) bool {
+	members, _ := membersFromStore(c.lg, c.v2store)
+	m := new(Member)
+	if err := json.Unmarshal(cc.Context, m); err != nil {
+		if c.lg != nil {
+			c.lg.Panic("failed to unmarshal member", zap.Error(err))
+		} else {
+			plog.Panicf("unmarshal member should never fail: %v", err)
+		}
+	}
+
+	if !m.IsLearner && members[types.ID(cc.NodeID)].IsLearner {
```

Comment:
Since we already have `m`, I think using `m.ID` instead of `types.ID(cc.NodeID)` is more readable?

Reply:
If we are adding a new regular member, `!m.IsLearner` is `true`, but `members[types.ID(cc.NodeID)]` does not exist?